src/test/java/htsjdk/samtools/SAMFileWriterFactoryTest.java

```diff
@@ -304,4 +320,75 @@ public void testAsync() throws IOException {
         writer = builder.setUseAsyncIo(false).makeWriter(header, false, outputFile, referenceFile);
         Assert.assertFalse(writer instanceof AsyncSAMFileWriter, "testAsync option=unset");
     }
+
+    @Test
+    public void testMakeWriterForSamExtension() throws IOException {
+        final File tmpFile = File.createTempFile("testMakeWriterForSamExtension", ".sam");
+        tmpFile.deleteOnExit();
+        final SAMFileWriter samFileWriter = new SAMFileWriterFactory().makeWriter(new SAMFileHeader(), true, tmpFile, null);
+        samFileWriter.close();
+
+        final FileInputStream fis = new FileInputStream(tmpFile);
+        for (byte b:"@HD\tVN:".getBytes()) {
+            Assert.assertEquals((byte)(fis.read() & 0xFF), b);
+        }
+        fis.close();
+    }
+
+    @Test
+    public void testMakeWriterForBamExtension() throws IOException {
+        final File tmpFile = File.createTempFile("testMakeWriterForBamExtension", ".bam");
```

Comment:
Can you update `BamFileIoUtils.BAM_FILE_EXTENSION` to use `amReader.Type.BAM_TYPE.fileExtension()`?  It makes me nervous we have two places (that I can see) having it hard-coded to ".bam"